SHARE-243	Refactoring - Adminarea - Special Projects | Featured Projects

IsolatedSys opened the pull request:
When clicking on name it redirected to a empty page - should be fixed (testet)
wrote tests
deleted mosaic view (locally on this admin page); 
added all flavors to choices in edit page

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

dmetzner reviewed (changes requested):
Please don't just tick the Checklist without really checking the stuff.
Phpstan reports 2 errors.

For some reason the PR can't be auto-deployed:
![image](https://user-images.githubusercontent.com/40868718/80838791-19410d80-8bfa-11ea-96a0-6e16253a5c3b.png)

can you please rebase the PR to the current develop?

mfuerni commented:
> For some reason the PR can't be auto-deployed:
```
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run update to update them.
[...]
Symfony operations: 1 recipe (8494733d1e321e241ca8fda2d3c3b4d9)
  - Configuring fr3d/ldap-bundle (>=v4.0.0): From auto-generated recipe
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!
!!  In BaseNode.php line 441:
!!
!!    Invalid configuration for path "fr3d_ldap": Notice: Undefined index: driver
!!
!!
!!  In Configuration.php line 68:
!!
!!    Notice: Undefined index: driver
!!
!!
!!
Script @auto-scripts was called via post-install-cmd
>>> Failed to run composer install for pr531
```

dmetzner commented on src/Admin/FeaturedProgramAdmin.php:
```diff
@@ -145,11 +147,21 @@ protected function configureFormFields(FormMapper $formMapper): void
       ->add('file', FileType::class, $file_options)
       ->add('program_id', TextType::class, ['mapped' => false, 'data' => $id_value])
-      ->add('flavor')
+      ->add('flavor', ChoiceFieldMaskType::class, [
```
It would be really nice to generate the choices array automatically - else it will need work every time we add a new flavor. Something that should be possible with a few clicks in the admin interface in the future.

dmetzner replied:
@leno12 already did it in his PR. You might wanna take a look at it.
![image](https://user-images.githubusercontent.com/40868718/81038218-8d530e00-8ea5-11ea-93e8-4ada4adeb342.png)

IsolatedSys replied:
Thanks, already done. Now I work on your other requests.

dmetzner commented on tests/behat/features/admin/featured_programs.feature:
```diff
@@ -0,0 +1,128 @@
+      Then I should see "Edit \"program 1 (#1337-c0ffee)\""
+
+
+
```
A test or two for adding a new feature project would be nice. Failing + Passing

dmetzner reviewed (changes requested):
- remove sort by **id, flavor**
- if possible add tests for adding feature projects
- themes array would be nice if it was created dynamically

Still, it is just the admin interface if something requires too much work, just say so - nothing of the mentioned stuff is really a must have.

IsolatedSys commented:
- removed sort by id & flavor
- added tests for adding feature projects (fail, success and wrong image)
- flavor array created dynamically

dmetzner reviewed (changes requested):
When updating featured projects. Can you remove the ACL button? I see no reason for this button in this action.
![image](https://user-images.githubusercontent.com/40868718/81113645-df854500-8f20-11ea-8cb2-7cf4cadf9174.png

IsolatedSys committed:
SHARE-243 Refactoring - Adminarea - Special Projects | Featured Projects

When clicking on name it redirected to a empty page - should be fixed (testet); wrote tests, deleted mosaic view; in edit page, flavors are in a dropdownmenu; added all flavors to choices in edit page
composer updated & PHPStan error fixed
choices dynamically filled
id/flavor sorting disabled
wrote more tests
removed ACL button
